clients/src/main/java/org/apache/kafka/common/record/EosLogEntry.java

```diff
@@ -33,36 +32,36 @@
 import static org.apache.kafka.common.record.Records.LOG_OVERHEAD;
 
 public class EosLogEntry extends AbstractLogEntry implements LogEntry.MutableLogEntry {
-    static final int OFFSET_OFFSET = 0;
-    static final int OFFSET_LENGTH = 8;
-    static final int SIZE_OFFSET = OFFSET_OFFSET + OFFSET_LENGTH;
+    static final int BASE_OFFSET_OFFSET = 0;
+    static final int BASE_OFFSET_LENGTH = 8;
+    static final int SIZE_OFFSET = BASE_OFFSET_OFFSET + BASE_OFFSET_LENGTH;
     static final int SIZE_LENGTH = 4;
     static final int CRC_OFFSET = SIZE_OFFSET + SIZE_LENGTH;
     static final int CRC_LENGTH = 4;
     static final int MAGIC_OFFSET = CRC_OFFSET + CRC_LENGTH;
     static final int MAGIC_LENGTH = 1;
     static final int ATTRIBUTES_OFFSET = MAGIC_OFFSET + MAGIC_LENGTH;
-    static final int ATTRIBUTE_LENGTH = 1;
-    static final int TIMESTAMP_OFFSET = ATTRIBUTES_OFFSET + ATTRIBUTE_LENGTH;
-    static final int TIMESTAMP_LENGTH = 8;
-    static final int OFFSET_DELTA_OFFSET = TIMESTAMP_OFFSET + TIMESTAMP_LENGTH;
-    static final int OFFSET_DELTA_LENGTH = 4;
-    static final int PID_OFFSET = OFFSET_DELTA_OFFSET + OFFSET_DELTA_LENGTH;
+    static final int ATTRIBUTE_LENGTH = 2;
+    static final int BASE_TIMESTAMP_OFFSET = ATTRIBUTES_OFFSET + ATTRIBUTE_LENGTH;
```

Comment:
I felt the use of "first" was misleading since after compaction, it may no longer be first (we keep around the offset and sequence number from the original message set). There doesn't seem to be a good reason necessarily to do the same for timestamps, but I felt we might as well be consistent with naming.